include/subsystems/odometry/odometry_nwheel.h

```diff
@@ -0,0 +1,244 @@
+#pragma once
+#undef __ARM_NEON__
+#undef __ARM_NEON
+#include "../core/include/subsystems/custom_encoder.h"
+#include "../core/include/subsystems/odometry/odometry_base.h"
+#include "../core/include/utils/math_util.h"
+#include <Eigen/Dense>
+
+/**
+ * OdometryNWheel
+ *
+ * This class handles the code for an N-pod odometry setup, where there are N <WHEELS> free spinning omni wheels
+ * (dead wheels) placed in any known configuration on the robot.
+ *
+ * Example of a possible wheel configuration:
+ *
+ *  +Y   ---------------
+ *  ^    |    ===      |
+ *  |    |             |
+ *  |    | ||   O      |
+ *  |    | ||          |
+ *  |    |      ===    |
+ *  |    ---------------
+ *  |
+ *  +-------------------> + X
+ *
+ * Where O is the center of rotation. The robot will monitor the changes in rotation of these wheels, use this to
+ * calculate a pose delta, then integrate the deltas over time to determine the robot's position.
+ *
+ * This is a "set and forget" class, meaning once the object is created, the robot will immediately begin
+ * tracking it's movement in the background.
+ *
```

Comment:
Can you add a reference to the notebook entry (`2024-2025 Notebook: Entries/Software Entries/Localization/N-Pod Odometry`) for all the poor souls who come across this later and have to understand it (me in like 3 days when i forget the math)